src/gcp_scanner/scanner.py

```diff
@@ -215,24 +166,25 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
         logging.error('Try removing the %s file and restart the scanner.',
                       output_file_name)
 
-      for crawler_name, client_name in crawl_client_map.items():
-        if is_set(scan_config, crawler_name):
-          crawler_config = {}
-          if scan_config is not None:
-            crawler_config = scan_config.get(crawler_name)
-          # add gcs output path to the config.
-          # this path is used by the storage bucket crawler as of now.
-          crawler_config['gcs_output_path'] = gcs_output_path
-          # crawl the data
-          crawler = CrawlerFactory.create_crawler(crawler_name)
-          client = ClientFactory.get_client(client_name).get_service(
-            credentials,
-          )
-          project_result[crawler_name] = crawler.crawl(
-            project_id,
-            client,
-            crawler_config,
-          )
+      results_crawl = dict()
+      with concurrent.futures.ProcessPoolExecutor() as executor:
```

Comment:
Am I right to assume that we create N number of processes here equal to number of resources required for scan?

We will definitely need to allow user to specify # of processes allowed to use by the scanner where 1 is default value

Reply:
Yup! by default it automatically creates processes equal to the number of Cores in the Machine.

So, we should add a parameter like `--parallel` or `-p` followed by the number of cores the person wants to assign.

Reply:
Yes, -p flag makes sense